src/dashboard/pages/login/login_page.py

```diff
@@ -0,0 +1,168 @@
+                        children=[
+                            html.H1(
+                                className=" mt-4 pd-4 ",
+                                children=[
+                                    "Welcome to GraphIt",
+                                ],
+                            ),
+                            # product logo
+                            html.Img(
+                                src=dash.get_asset_url("logoTransparent.png"),
+                            ),
+                        ],
+                    ),
+                    # error pop-up
+                    html.Div(
+                        id="error_input_message",
+                        className=(
+                            "hidden drop-shadow-2xl fixed bottom-[-100px] "
+                            "left-1/2 transform -translate-x-1/2 h-fit w-fit "
+                            "fit-content bg-red-100 border border-red-400 "
+                            "text-red-700 px-4 py-3 mt-3 rounded"
+                        ),
+                        children=[
+                            html.H2(
+                                className="text-center",
+                                children="Wrong username or password!",
+                            ),
+                        ],
+                    ),
+                ],
+            ),
+        ],
+    )
+
+
+@callback(
+    Output("dcc_location_login", "children"),
+    Output("error_input_message", "className"),
+    Output("username", "className"),
+    Output("password", "className"),
+    State("error_input_message", "className"),
+    State("username", "value"),
+    State("password", "value"),
+    Input("username", "n_submit"),
+    Input("password", "n_submit"),
+    Input("login_button", "n_clicks"),
+    prevent_initial_call=True,
+)
+def update_login(
+    error_pop_up: str, username: str, password: str, *_: int
+) -> tuple[None or dcc.Location, str, str, str]:
+    """Updates login page when inputing the password or username.
+
+    This function updates the username field and password field
+    style by updating their className, in which the text field
+    would have a red hue to mark that something when wrong.
+    The function would also make the error pop up not hidden by
+    changing the error_pop_up.
+
+    Args:
+        error_pop_up (str): The class name of error pop up element.
+        username (str): The class name of the username text field.
+        password (str): the class name of the passowrd text field.
+    """
+    # If it is the correct username and password that we change the page
+    # to the home page.
+    if username == "admin" and password == "admin":
+        return (
+            dcc.Location(pathname="", id="dcc_location_login"),
+            error_pop_up + " hidden",
+            NORMAL_FIELD_CLASS,
+            NORMAL_FIELD_CLASS + PASS_FIELD_EXTRA_CLASS,
+        )
+    else:
+        return (
+            None,
```

Comment:
Use `dash.no_update` to signal that a callback should not update one of its outputs. 
https://dash.plotly.com/advanced-callbacks

Reply:
Resolved :+1: